src/containers/my-cooperations/cooperation-details/CooperationDetails.tsx

```diff
@@ -160,18 +180,73 @@ const CooperationDetails = () => {
       ? cooperation.initiator
       : cooperation.receiver
 
-  const acceptClosingProcess = !isClosed && (
-    <AcceptCooperationClosing
-      isReasonSubmitted={false}
-      onAccept={handleCooperationClosingAccept}
-      onReasonSubmit={() => {}}
-      user={closeCooperationInitiator.firstName}
-    />
-  )
+  const closeCooperationReceiver =
+    cooperation.needAction.role === cooperation.initiatorRole
+      ? cooperation.initiator
+      : cooperation.receiver
 
-  const isCooperationClosingRequestSend =
-    cooperation.needAction.role === userRole &&
+  const roleBasedStatus =
+    cooperation.needAction.role === userRole
+      ? StatusEnum.NeedAction
+      : StatusEnum.RequestToClose
+
+  const cooperationStatus =
     cooperation.status === StatusEnum.RequestToClose
+      ? roleBasedStatus
+      : cooperation.status
+
+  const getCooperationClosingModal = () => {
+    const lastMessage = cooperation.needAction.messages.at(-1)
+    const secondLastMessage = cooperation.needAction.messages.at(-2)
+
+    if (cooperation.status !== StatusEnum.RequestToClose) {
+      return null
+    }
```

Comment:
Let's calculate messages only when we need them
```suggestion
    if (cooperation.status !== StatusEnum.RequestToClose) {
      return null
    }
    
    const lastMessage = cooperation.needAction.messages.at(-1)
    const secondLastMessage = cooperation.needAction.messages.at(-2)
```